changes/2024-TBD_cache-across-hierarchy-keyrings/change.md

```diff
@@ -43,49 +51,41 @@ By "KMS Relationship", we mean any or all of the following:
 - Credentials used when creating the KMS Client, and thus used when calling KMS
 - Other properties of the KMS Client, such as the region, or request headers
 
+## Background
 
-### Conventions used in this document
-
-The key words
-"MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
-"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
-in this document are to be interpreted as described in
-[RFC 2119](https://tools.ietf.org/html/rfc2119).
-
-## Summary
+A [Cryptographic Materials Cache](https://github.com/awslabs/aws-encryption-sdk-specification/blob/master/framework/cryptographic-materials-cache.md) can be used by a Hierarchy Keyring, a Caching CMM, and the DB-ESDK.
+It can store four types of [materials](https://github.com/aws/aws-cryptographic-material-providers-library/blob/c07a51fc29ff70411f7573bca96d2a091db8c1ed/AwsCryptographicMaterialProviders/dafny/AwsCryptographicMaterialProviders/Model/cryptographic-materials-cache.smithy#L89) for three different use-cases:
 
-To improve the MPL Consumer data key caching expierence,
-when using the Hierarchical Keyring,
-we need to allow caching across KeyStores/KMS Clients/KMS Keys.
+- Hierarchy Keyring: keyStore#BranchKeyMaterials
+- CachingCMM: EncryptionMaterials and DecryptionMaterials
+- DB-ESDK: keyStore#BeaconKeyMaterials
 
-To facilitate Caching across KeyStores/KMS Clients/KMS Keys,
-we MUST break the Cryptographic Materials Cache (CMC)
-out of the Hierarchy Keyring.
+These materials have certain cache identifiers for accessing
+them until a TTL. As of [today](#today), the cache identifiers
+are set up such that:
 
-By allowing MPL Consumers to provide an already initialized CMC
-to the Hierarchy Keyring at construction,
-the CMC MAY cache Branch Keys protected by different
-KMS Relationships.
+- Only one Hierarchy Keyring can use one cache,
+  i.e., currently, we do NOT support a shared cache
+  between multiple Hierarchy Keyrings.
+- Multiple CachingCMMs can use a single shared cache
+  (this is for native languages, there is currently no
+  CachingCMM implemented in dafny)
+- We do NOT support a shared cache between multiple
+  DB-ESDK clients.
```

Comment:
Issue/Suggestion/Blocking: The DB-ESDK client can, theoretically, consume CachingCMMs and Hierarchy Keyrings.

Where you write DB-ESDK, can you replace it with Searchable Encryption?

```suggestion
- We do NOT support a shared cache between multiple
  Searchable Encryption Configurations.
```

Reply:
I did not know that. Thanks for pointing this out.

Reply:
Updated this in the doc wherever I mentioned DB-ESDK